### PR TITLE
[FIX] account: avoid rounding error by not doing anything with empty …

### PR DESCRIPTION
**Pull request by jco-odoo**
…fiscal positions

opw-2447155

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jco-odoo** [comment]
Somehow the rounding for tax included is applied differently when the fiscal position applies even if there are not taxes in it.  This fix is too much monkey patch.  It would be better to find the source of the rounding error.  @smetl

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -2835,7 +2835,7 @@ def _set_price_and_tax_after_fpos(self):
         # adapt the price_unit to the new tax.
         # E.g. mapping a 10% price-included tax to a 20% price-included tax for a price_unit of 110 should preserve
         # 100 as balance but set 120 as price_unit.
-        if self.tax_ids and self.move_id.fiscal_position_id:
+        if self.tax_ids and self.move_id.fiscal_position_id and self.move_id.fiscal_position_id.tax_ids:
```
Comment (smetl) [inline addons/account/models/account_move.py:2838]: replace `business_vals = self._get_fields_onchange_balance(balance=balance)` by `business_vals = self._get_fields_onchange_balance(balance=balance, price_subtotal=price_subtotal)` instead.

2.4 / 1.14 = 2.105263158 ~= 2.11 but 2.11 * 1.14 = 2.4054 ~= 2.41

Comment (smetl) [reply addons/account/models/account_move.py:2838]: EDIT: Forget about this: this is a bug in the compute_all method with `handle_price_include`. I will fix it tomorrow

**Comment by smetl** [comment]
2.4 / 1.14 = 2.105263158 ~= 2.11 but 2.11 * 1.14 = 2.4054 ~= 2.41

I'm ok with your fix after all.
@robodoo r+

**Pull request closed without merging**